sympy/tensor/tensor.py

```diff
@@ -2110,7 +2110,7 @@ def contract_and_permute(metric, array, pos):
 
         return free_ind2, array
 
-    def replace_with_arrays(self, replacement_dict, indices):
+    def replace_with_arrays(self, replacement_dict, indices=[]):
```

Comment:
@oscargus , is this one of those cases where the default should be None and inside the code `indices = indices or []` should be written?

Reply:
I do not think so since `indices` is not (does not looks like it is) modified. I didn't check all the methods that uses indices though, so it may be safest to do it anyway.

@kangzhiq it turns out that if `indices` is modified at some stage in the code, the default value will keep those changes for the next time the method is called (as the list object is only instantiated once). Hence, it may be safest to use a `None` default and `indices = indices or []` as the first line.

Reply:
I would always avoid using a mutable default for a function parameter even if it isn't modified anywhere. Seeing it automatically makes me suspicious so that I then check that it is not modified anywhere. Also you have to ensure that no reference to the list is returned from the function since it can be modified by the caller afterwards as well.